### PR TITLE
Feature/fix input

### PR DESCRIPTION
**Pull request by Dartemiss**
What's new?
- Refactor input to input manager on scripts
- Refactor controller handling
- Fix critical animation bugs

### CODE REVIEW

File: Engine/Module/ModuleDebugDraw.cpp [hunk 1]
```diff
@@ -499,42 +500,35 @@ void ModuleDebugDraw::RenderLightGizmo() const
...
+						color
+					);
+				}
+
+				if (color.x == 1.f)
+					color = float3(0.f, 1.f, 0.f);
+				else if (color.y == 1.f)
+					color = float3(0.f, 0.f, 1.f);
+				else if (color.z == 1.f)
+					color = float3(1.f, 0.f, 0.f);
```
Comment (JoanStinson) [inline Engine/Module/ModuleDebugDraw.cpp:587]: This is supposed to be like this?

**Review by JoanStinson — APPROVED** [review]
gg 👍

**Comment by mariofv** [comment]
> And with this we need the <= or not?

@yalania We dont need it, changing the +1 in the AnimationController solved the issue

File: Engine/Module/ModuleDebugDraw.cpp (hunk 1 above)
Comment (mariofv) [reply Engine/Module/ModuleDebugDraw.cpp:587]: Yes

File: Engine/Module/ModuleInput.h [hunk 2]
```diff
@@ -518,19 +532,19 @@ class ModuleInput : public Module
 	Uint8 mouse_clicks;
 	bool mouse_moving;
 
-	float2 left_joystick[MAX_PLAYERS];
-	float2 right_joystick[MAX_PLAYERS];
+	//float2 left_joystick[MAX_PLAYERS];
+	//float2 right_joystick[MAX_PLAYERS];
```
Comment (mariofv) [inline Engine/Module/ModuleInput.h]: In general, delete commented useless code please 😈

File: Engine/Module/ModuleDebugDraw.cpp (hunk 1 above)
Comment (mariofv) [reply Engine/Module/ModuleDebugDraw.cpp:587]: Actually, it can be improved. Doing it right now!

**Review by mariofv — APPROVED** [review]
GG EZ

**Pull request merged**